pkg/controller/kubedirectorcluster/cluster.go

```diff
@@ -308,15 +310,9 @@ func (r *ReconcileKubeDirectorCluster) syncCluster(
 		return configMetaErr
 	}
 	if state == clusterMembersChangedUnready || (currentHash != cr.Status.LastConnectionHash) {
-		if cr.Status.SpecGenerationToProcess == nil {
-			initSpecGen := int64(1)
-			cr.Status.SpecGenerationToProcess = &initSpecGen
-		}
-		if currentHash != cr.Status.LastConnectionHash {
-			incremented := atomic.AddInt64(cr.Status.SpecGenerationToProcess, 1)
-			cr.Status.SpecGenerationToProcess = &incremented
-			cr.Status.LastConnectionHash = currentHash
-		}
+		incremented := *cr.Status.SpecGenerationToProcess + int64(1)
```

Comment:
does it matter atomic increment v/s this? Just trying to understand the pros and cons.

Reply:
No other goroutine is reading or writing this value so I don't think we need to incur the overhead of using atomic operations. Just generally speaking, we don't have parallel processes that are communicating through any shared-memory data structures.

Reply:
yes makes sense 👍